static/js/hash_util.js

```diff
@@ -273,7 +273,7 @@ export function is_spectator_compatible(hash) {
     const web_public_allowed_hashes = [
         "",
         "narrow", // full #narrow hash handled in narrow.is_spectator_compatible
-        "recent_topics",
+        "recent",
```

Comment:
I wasn't sure if we need to keep "recent_topics" here with the `TODO/compatibility` note as well.

Reply:
It's probably a good idea; might not come up in practice but it's cheap to do.

Reply:
I think we do need to leave this for now, but it can be removed once one cannot upgrade directly from Zulip 5.x, because at that point the only situation with the "#recent_topics" has that we'll need to handle is from the Welcome Bot messages. The reload logic won't be needed.

I added a very short `TODO/compatibility` comment for this; adding a longer comment wasn't as readable, but we might want to try anyway?

Reply:
This is why we generally prefer to avoid the inline comments, rather than comments on the line before what we're commenting. I rewrote this as a proper paragraph comment as part of merging this.